packages/notifications/push/amplify_push_notifications/android/src/main/kotlin/com/amazonaws/amplify/amplify_push_notifications/AmplifyPushNotificationBroadcastReceiver.kt

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.amplify.amplify_push_notifications
+
+
+import android.annotation.SuppressLint
+import android.app.ActivityManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import com.amplifyframework.pushnotifications.pinpoint.utils.PushNotificationsUtils
+import com.google.firebase.messaging.RemoteMessage
+
+// TODO: This BroadcastReceiver needs to be replaced by the Firebase Service
+class PushNotificationReceiver : BroadcastReceiver() {
+    companion object {
+        private val TAG = "PushNotificationReceiver"
+    }
+
+    @SuppressLint("LongLogTag")
+    override fun onReceive(context: Context, intent: Intent?) {
+        intent?.let {
+            if (isPushNotificationIntent(it)) {
```

Comment:
```suggestion
            if (!isPushNotificationIntent(it)) {
                return
            }
```

Prefer exit early whenever possible